scratch/copy-mk.py

```diff
@@ -66,6 +83,14 @@ def parse_args(argv: Sequence[str] | None = None) -> argparse.Namespace:
     return args
 
 
+def _assemble_regex(names: Sequence[str]) -> str:
+    """Assemble a regex from a sequence of strings."""
+    regex = "|".join(re.escape(name) for name in names)
+    if not regex:
+        return {}  # Would get an error trying to query an empty regex
```

Comment:
That's not a valid return type for this function (I'm surprised mypy didn't tell you off - maybe it's not checking scratch/). Rather keep this logic in the caller.